rust/connlib/clients/android/src/lib.rs

```diff
@@ -381,11 +397,13 @@ fn connect(
         sockets,
         private_key,
         Some(os_version),
-        callback_handler,
+        callback_handler.clone(),
         Some(MAX_PARTITION_TIME),
         runtime.handle().clone(),
     );
 
+    callback_handler.session = Some(session.clone());
```

Comment:
This isn't gonna work because the instance of `CallbackHandler` that got passed into `Session` doesn't have the `Session` set :man_facepalming: 

Need to think of something else here.